test_report/tests/report.rs

```diff
@@ -0,0 +1,97 @@
+use prost::Message;
+use prost_wkt_types::Timestamp;
+use proto::test_context::test_run::TestCaseRunStatus;
+use tempfile::tempdir;
+use test_report::report::{MutTestReport, Status};
+use test_utils::mock_git_repo::setup_repo_with_commit;
+use test_utils::mock_server::MockServerBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn publish_test_report() {
+    let temp_dir = tempdir().unwrap();
+    let repo_setup_res = setup_repo_with_commit(&temp_dir);
+    assert!(repo_setup_res.is_ok());
+    let set_current_dir_res = std::env::set_current_dir(&temp_dir);
+    assert!(set_current_dir_res.is_ok());
+    let state = MockServerBuilder::new().spawn_mock_server().await;
+    std::env::set_var("TRUNK_PUBLIC_API_ADDRESS", &state.host);
+    std::env::set_var("CI", "1");
+    std::env::set_var("GITHUB_JOB", "test-job");
+    std::env::set_var("TRUNK_API_TOKEN", "test-token");
+    std::env::set_var("TRUNK_ORG_URL_SLUG", "test-org");
+
+    let thread_join_handle = std::thread::spawn(|| {
+        let test_report = MutTestReport::new("test".into());
+        test_report.add_test(
+            Some("1".into()),
+            "test-name".into(),
+            "test-classname".into(),
+            "test-file".into(),
+            "test-parent-name".into(),
+            None,
+            Status::Success,
+            0,
+            0,
+            0,
+            "test-message".into(),
+        );
+        let result = test_report.publish(".".into());
+        assert_eq!(result, true);
+    });
+    let thread_res = thread_join_handle.join();
+    assert!(thread_res.is_ok());
+}
+
+#[test]
+fn save_test_report() {
+    let temp_dir = tempdir().unwrap();
+    setup_repo_with_commit(&temp_dir).unwrap();
+    let _ = std::env::set_current_dir(&temp_dir);
+    let test_report = MutTestReport::new("test".into());
```

Comment:
IMO, we should inspect the bundle that was uploaded in the first test to be more E2E than to look at the intermediary data like this test does

Reply:
Done.